cli/diff.rs

```diff
@@ -3,6 +3,144 @@
 use crate::colors;
 use dissimilar::{diff as difference, Chunk};
 
+/// Print diff of the same file_path, before and after formatting.
+///
+/// Diff format is loosely based on Github diff formatting.
+pub fn diff(orig_text: &str, edit_text: &str) -> String {
+  if orig_text == edit_text {
+    return String::new();
+  }
+
+  // normalize newlines as it adds too much noise if they differ
+  let orig_text = orig_text.replace("\r\n", "\n");
+  let edit_text = edit_text.replace("\r\n", "\n");
+
+  if orig_text == edit_text {
+    return " | Text differed by line endings.\n".to_string();
+  }
+
+  DiffBuilder::build(&orig_text, &edit_text)
+}
+
+struct DiffBuilder {
+  output: String,
+  line_number_width: usize,
+  orig_line: usize,
+  edit_line: usize,
+  orig: String,
+  edit: String,
+  has_changes: bool,
+}
```

Comment:
Unfortunately I had to change the function to a struct in order to make the borrow checker happy and not duplicate a lot of code with the new `flush_changes` method.